OffchainLabs/arbitrum-token-bridge
# feat: ETH transfers to a custom address

Features:
- ETH transfers to a custom address (both EOA and SC)
- Improved UX when dealing with destination address

Transfers to custom destination address:
- [x] Deposits EOA (ETH)
- [x] Withdrawals EOA (ETH)
- [x] Deposits Smart Contract (ETH)
- [x] Withdrawals Smart Contract (ETH)
- [x] Deposits EOA (ERC-20)
- [x] Withdrawals EOA (ERC-20)
- [x] Deposits Smart Contract (ERC-20)
- [x] Withdrawals Smart Contract (ERC-20)

Transfers to the same destination address:
- [x] Deposits (ETH)
- [x] Withdrawals (ETH)
- [x] Deposits (ERC-20)
- [x] Withdrawals (ERC-20)

**Testing instructions**
EOA - just provide a different destination address, funds should land there.

Contract wallet:
Safe doesn't support Arbitrum Goerli. We have an internal testing environment for this.

General Safe instructions:
1. Go to [Safe](https://app.safe.global/) and connect to the bridge using WalletConnect (don't close this window, new tx's will appear here).
2. Insert destination address of the wallet you send the funds to. (If depositing, it's Arb1 wallet, if withdrawing, Goerli).
3. Approve tx in Safe.

**Note that to claim withdrawals you need to be connected to L1 with the same address as L2 sender**
